api/v3/Case.php

```diff
@@ -77,42 +77,44 @@ function civicrm_api3_case_create($params) {
     // Update an existing case
     // FIXME: Some of this logic should move to the BAO object?
     // FIXME: Should we check if case with ID actually exists?
-    if (!isset($params['case_id']) && isset($params['id'])) {
-      $params['case_id'] = $params['id'];
-    }
 
     if (array_key_exists('creator_id', $params)) {
       throw new API_Exception('You cannot update creator id');
     }
 
-    $mergedCaseId = $origContactIds = array();
+    $mergedCaseIds = $origContactIds = array();
 
-    // get original contact id and creator id of case
+    // If a contact ID is specified we need to make sure this is the main contact ID for the case (and update if necessary)
     if (!empty($params['contact_id'])) {
       $origContactIds = CRM_Case_BAO_Case::retrieveContactIdsByCaseId($params['id']);
-      $origContactId = CRM_Utils_Array::first($origContactIds);
-    }
 
-    // FIXME: Refactor as separate method to get contactId
-    if (count($origContactIds) > 1) {
-      // check valid orig contact id
-      if (empty($params['orig_contact_id'])) {
-        throw new API_Exception('Case is linked with more than one contact id. Provide the required params orig_contact_id to be replaced');
+      // Get the original contact ID for the case
+      // FIXME: Refactor as separate method to get contactId
+      if (count($origContactIds) > 1) {
+        // Multiple original contact IDs. Need to specify which one to use as a parameter
+        if (empty($params['orig_contact_id'])) {
+          throw new API_Exception('Case is linked with more than one contact id. Provide the required params orig_contact_id to be replaced');
+        }
+        if (!empty($params['orig_contact_id']) && !in_array($params['orig_contact_id'], $origContactIds)) {
+          throw new API_Exception('Invalid case contact id (orig_contact_id)');
+        }
+        $origContactId = $params['orig_contact_id'];
       }
-      if (!empty($params['orig_contact_id']) && !in_array($params['orig_contact_id'], $origContactIds)) {
-        throw new API_Exception('Invalid case contact id (orig_contact_id)');
+      else {
+        // Only one original contact ID
+        $origContactId = CRM_Utils_Array::first($origContactIds);
       }
-      $origContactId = $params['orig_contact_id'];
-    }
 
-    // check for same contact id for edit Client
-    if (!empty($params['contact_id']) && !in_array($params['contact_id'], $origContactIds)) {
-      $mergedCaseId = CRM_Case_BAO_Case::mergeCases($params['contact_id'], $params['case_id'], $origContactId, NULL, TRUE);
-    }
+      // Get the specified main contact ID for the case
+      $mainContactId = CRM_Utils_Array::first($params['contact_id']);
```

Comment:
@mattwire  What guarantees that this value is an array?

Reply:
@colemanw This line: https://github.com/civicrm/civicrm-core/pull/11830/files#diff-6becb88c3b67ad08795846ea0642d97dR714